internal/processing/status/status.go

```diff
@@ -3,21 +3,24 @@ package status
 import (
 	"fmt"
 	"github.com/kyma-project/api-gateway/apis/gateway/v1beta1"
-	"github.com/kyma-project/api-gateway/apis/gateway/versions"
+	"github.com/kyma-project/api-gateway/apis/gateway/v2alpha1"
 	"github.com/kyma-project/api-gateway/internal/validation"
 )
 
 type ReconciliationStatus interface {
-	UpdateStatus(status *v1beta1.APIRuleStatus) error
+	UpdateStatus(status Status) error
```

Comment:
what do you think of using `any` instead of middle struct that is holding two references. this will make simpler usage of the func and requires only reflect casting check in the respective version function, e.g.:
```
func (s ReconciliationV1beta1Status) UpdateStatus(status any) error {
	st, ok := status.(gatewayv1beta1.APIRuleStatus)
	if !ok {
		return errors.New("status is not of type gatewayv1beta1.APIRuleStatus")
	}
```